t/25-downloader.t

```diff
@@ -68,28 +68,34 @@ sub stop_server {
 }
 
 my $mojo_tmpdir = tempdir;
-my $downloader  = OpenQA::Downloader->new(log => $log, sleep_time => 1, tmpdir => $mojo_tmpdir);
-my $tempdir     = tempdir;
-my $to          = $tempdir->child('test.qcow');
+my $sleep_time  = 0.05;
+my $downloader = OpenQA::Downloader->new(log => $log, sleep_time => $sleep_time, attempts => 3, tmpdir => $mojo_tmpdir);
+my $ua         = $downloader->ua;
+my $tempdir    = tempdir;
+my $to         = $tempdir->child('test.qcow');
 
 subtest 'Connection refused' => sub {
     my $from = "http://$host/tests/922756/asset/hdd/sle-12-SP3-x86_64-0368-textmode@64bit.qcow2";
+    $ua->connect_timeout(0)->inactivity_timeout(0);
     ok !$downloader->download($from, $to), 'Failed';
 
     ok !-e $to, 'File not downloaded';
 
-    like $cache_log, qr/Downloading "test.qcow" from "$from"/,                               'Download attempt';
-    like $cache_log, qr/Download of "$to" failed: 521/,                                      'Real error is logged';
-    like $cache_log, qr/Download error 521, waiting 1 seconds for next try \(4 remaining\)/, '4 tries remaining';
-    like $cache_log, qr/Download error 521, waiting 1 seconds for next try \(3 remaining\)/, '3 tries remaining';
-    like $cache_log, qr/Download error 521, waiting 1 seconds for next try \(2 remaining\)/, '2 tries remaining';
-    like $cache_log, qr/Download error 521, waiting 1 seconds for next try \(1 remaining\)/, '1 tries remaining';
+    like $cache_log, qr/Downloading "test.qcow" from "$from"/, 'Download attempt';
+    like $cache_log, qr/Download of "$to" failed: 521/,        'Real error is logged';
+    like $cache_log, qr/Download error 521, waiting $sleep_time seconds for next try \(2 remaining\)/,
+      '2 tries remaining';
+    like $cache_log, qr/Download error 521, waiting $sleep_time seconds for next try \(1 remaining\)/,
+      '1 tries remaining';
+    unlike $cache_log, qr/Download error 521, waiting $sleep_time seconds for next try \(3 remaining\)/,
+      'only 3 attempts';
```

Comment:
style nit-pick: To prevent the line breaks as we are already using regex's remove some part from the middle, e.g. `qr/Download error 521, waiting $sleep_time.*\(2 remaining\)/